src/main/java/edu/ucsb/cs156/rec/controllers/RecommendationRequestController.java

```diff
@@ -0,0 +1,92 @@
+    @GetMapping("/all")
+    public Iterable<RecommendationRequest> allRecommendationRequests() {
+        Iterable<RecommendationRequest> recommendationRequests = recommendationRequestRepository.findAll();
+        return recommendationRequests;
+    }
+
+    /**
+     * This method returns a single recommendationrequests.
+     * @param id id of the recommendationrequests to get
+     * @return a single recommendationrequests
+     */
+    @Operation(summary = "Get a single recommendation request")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("")
+    public RecommendationRequest getById(
+            @Parameter(name = "id") @RequestParam Long id) {
+            RecommendationRequest recommendationRequest = recommendationRequestRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(RecommendationRequest.class, id));
+
+        return recommendationRequest;
+    }
+
+    /**
+     * This method creates a new recommendationrequests. Accessible only to users with the role "ROLE_ADMIN" until our roles can be configured.
+     * @param professorName professor name of request
+     * @param professorEmail professor email of request
+     * @param recommendationTypes recommendation types of request
+     * @param details details of request
+     * @param submissionDate submission date of request
+     * @return the save recommendationrequests (with it's id field set by the database)
+     */
+    @Operation(summary = "Create a new recommendationrequests")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping("/post")
+    public RecommendationRequest postRecommendationRequests(
+            @Parameter(name = "professorName") @RequestParam String professorName,
+            @Parameter(name = "professorEmail") @RequestParam String professorEmail,
+            @Parameter(name = "recommendationTypes") @RequestParam String recommendationTypes,
+            @Parameter(name = "details") @RequestParam String details,
+            @Parameter(name = "submissionDate") @RequestParam LocalDateTime submissionDate)
+            {
+        //get current date right now and set status to pending
+        CurrentUser currentUser = getCurrentUser();
+        RecommendationRequest recommendationRequest = new RecommendationRequest();
+
+        recommendationRequest.setProfessorName(professorName);
+        recommendationRequest.setProfessorEmail(professorEmail);
+        recommendationRequest.setRequesterName(currentUser.getUser().getFullName());
+        recommendationRequest.setUser(currentUser.getUser());
+        recommendationRequest.setRecommendationTypes(recommendationTypes);
+        recommendationRequest.setDetails(details);
+        recommendationRequest.setStatus("PENDING");
+        // ideally would like this to just directly call LocalDateTime.now() cannot properly mock test this behavior with full coverage
+        recommendationRequest.setSubmissionDate(submissionDate);
```

Comment:
Yes you can: see https://www.baeldung.com/java-override-system-time

But, there's a better way. Use this: https://ucsb-cs156.github.io/topics/spring_boot/spring_boot_database.html#timestamps-on-database-rows